airbyte-cdk/bulk/core/load/src/test/kotlin/io/airbyte/cdk/task/DestinationTaskLauncherTest.kt

```diff
@@ -243,11 +248,28 @@ class DestinationTaskLauncherTest {
 
     class MockBatch(override val state: Batch.State) : Batch
 
+    @Singleton
+    @Requires(env = ["DestinationTaskLauncherTest"])
+    class MockExceptionHandler : DestinationTaskLauncherExceptionHandler {
+        val syncFailureChannel = Channel<Throwable>(Channel.UNLIMITED)
```

Comment:
for my understanding: this could just as well be an arraylist + mutex? i.e. we're not doing anything that strictly requires a channel, it's just more convenient?